cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js

```diff
@@ -446,6 +446,86 @@ class HydratorPlusPlusNodeConfigCtrl {
     }
     return true;
   }
+  validatePluginPreferences() {
+    const nodeInfo = this.state.node;
+    let schemaParseError = null;
+    let vm = this;
+    vm.validating = true;
+    const pluginInfo = angular.copy(nodeInfo.plugin);
+    pluginInfo.type = nodeInfo.type;
+
+    const requestBody = {
+      stage: {
+        name: nodeInfo.name,
+        plugin: pluginInfo,
+      },
+      inputSchemas: !nodeInfo.inputSchema ? [] : nodeInfo.inputSchema.map((input) => {
+        let schema;
+        try {
+          schema = JSON.parse(input.schema);
+        } catch (e) {
+          schemaParseError = e;
+        }
+        return {
+          stage: input.name,
+          schema
+        };
+      })
+    };
+
+    if (schemaParseError) {
+      vm.validationErrors = [schemaParseError];
+      return;
+    }
+
+    const parseResSchema = (res) => {
+      if (res.name && res.type && res.fields) {
+        return [this.HydratorPlusPlusNodeService.getOutputSchemaObj(res)];
+      }
+      let schemaArr = [];
+      angular.forEach(res, (value, key) => {
+        if (value.name && value.type && value.fields) {
+          schemaArr.push(this.HydratorPlusPlusNodeService.getOutputSchemaObj(value, key));
+        }
+      });
+      let recordSchemas = schemaArr.filter(schema => schema.name.substring(0, 6) === 'record');
```

Comment:
i'm not sure I understand what this is trying to do?

Reply:
I followed the code structure in output-schema.js, This is handling a case where outputSchema or portSchemas is an array and sorting based on if they are record schemas or not.
https://github.com/cdapio/cdap/blob/b31d0c456088719e7bfbaad67b9973b8441ae523/cdap-ui/app/directives/plugin-functions/functions/output-schema/output-schema.js#L70